akka-stream/src/main/scala/akka/stream/impl/Sources.scala

```diff
@@ -136,3 +139,59 @@ private[akka] final class SourceQueueAdapter[T](delegate: SourceQueue[T]) extend
   def offer(elem: T): CompletionStage[QueueOfferResult] = delegate.offer(elem).toJava
   def watchCompletion(): CompletionStage[Done] = delegate.watchCompletion().toJava
 }
+
+/**
+ * INTERNAL API
+ */
+private[stream] final class BlockingStreamSource[T, S](create: () ⇒ S,
```

Comment:
As the original author I had to re-read the ticket :-) I agree that in addition to the blocking resource there should also be a variant for a non-blocking/async resource.